invenio_stats/aggregations.py

```diff
@@ -97,13 +99,26 @@ def wrapped(self, *args, **kwargs):
 
         return wrapped
 
+    def prepare_bookmark(self, value=None):
+        """Prepare the value that will be set by the bookmark."""
+        if value:
+            self.new_timestamp = value
+        else:
+            # By default, it is the current timestamp, minus 1 minute, since indices might take a bit to be refreshed
+            self.new_timestamp = (datetime.utcnow() - timedelta(minutes=1)).strftime(
```

Comment:
disclaimer: I don't have a lot of context on this, so just dropping my thoughts.
the one minute seems a little bit flaky, we struggled also in the UI to come up with a correct refresh delta, and it seems quite random. Is there any way to make it resilient to random refresh time of indices?

Reply:
I have seen the one minute refresh rate in some of the indices in opensearch, and that's why I took that number. 

A different option would be to set it to a percentage of time of how often the collector runs, like let's say 10%. So, if the collector runs ever hour, it will reprocess the last six minutes. Would that be better?

Reply:
After discussing also with Nicola, we left the 1 minute delay. It has been moved to the retrieval of the bookmark, instead of the setting.